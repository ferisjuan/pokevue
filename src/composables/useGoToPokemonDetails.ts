import { useRouter } from "vue-router";

export const useGoToPokemonDetails = () => {
  const router = useRouter();
  const goToPokemonDetails = (id: number) =>
    router.push({
      name: "pokemon-details",
      params: { id: String(id) },
    });
  return { goToPokemonDetails };
};
