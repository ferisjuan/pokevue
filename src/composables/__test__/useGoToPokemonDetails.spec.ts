import { expect, describe, it, vi } from "vitest";
import { useGoToPokemonDetails } from "../useGoToPokemonDetails";

vi.mock("vue-router", () => ({
  useRouter: () => ({
    push: vi.fn(),
  }),
}));

describe("useGoToPokemonDetails", () => {
  it("should return a function", () => {
    expect(typeof useGoToPokemonDetails()).toBe("object");
  });

  it("should navigate to the pokemon details page", () => {
    const { goToPokemonDetails } = useGoToPokemonDetails();
    expect(goToPokemonDetails(1)).toMatchSnapshot();
  });
});
