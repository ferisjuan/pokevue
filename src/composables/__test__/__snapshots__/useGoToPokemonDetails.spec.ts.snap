// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useGoToPokemonDetails > should navigate to the pokemon details page 1`] = `undefined`;
